pkg/cloudprovider/cloudprovider.go

```diff
@@ -108,6 +118,21 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 }
 
+func resolveClusterEndpoint(ctx context.Context, eksAPI eksiface.EKSAPI) (string, error) {
+	resolveClusterEndpointFromSettings := settings.FromContext(ctx).ClusterEndpoint
+	if resolveClusterEndpointFromSettings != "" {
+		return resolveClusterEndpointFromSettings, nil // cluster endpoint is explicitly set
+	}
+	clusters, err := eksAPI.DescribeCluster(&eks.DescribeClusterInput{
```

Comment:
```suggestion
	out, err := eksAPI.DescribeCluster(&eks.DescribeClusterInput{
```

nit, I think only one cluster is returned here